src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -432,7 +432,7 @@ internal ChangeSignatureResult ChangeSignatureWithContext(ChangeSignatureAnalysi
             foreach (var docId in nodesToUpdate.Keys)
             {
                 var updatedDoc = currentSolution.GetRequiredDocument(docId).WithSyntaxRoot(updatedRoots[docId]);
-                var docWithImports = ImportAdder.AddImportsFromSymbolAnnotationAsync(updatedDoc, safe: true, cancellationToken: cancellationToken).WaitAndGetResult(cancellationToken);
+                var docWithImports = ImportAdder.AddImportsFromSymbolAnnotationAsync(updatedDoc, cancellationToken: cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
the realization of this change is that all callers (outside of tests) passed in `safe: true` for `AddImportsFromSymbolAnnotation`, and `safe: false` for `AddImportsFromSyntaxes`.  So instead of having this duplicative flag, the behavior is just determined by which of these you are calling.

I then updated the tests to not call into codepaths that the product itself cannot hit.